src/test/java/com/redhat/devtools/lsp4ij/fixtures/LSPFoldingRangeFixtureTestCase.java

```diff
@@ -48,12 +50,40 @@ protected LSPFoldingRangeFixtureTestCase(String... fileNamePatterns) {
         super(fileNamePatterns);
     }
 
+    /**
+     * Verifies that the folding ranges are as-expected by matching them against expected
+     * <code>&lt;start<i>N</i>&gt;</code> / <code>&lt;end<i>N</i>&gt;</code> tag pairs that represent the start/end
+     * offsets of the corresponding {@link FoldRegion fold regions}.
+     *
+     * @param fileName                       the file name
+     * @param fileBody                       the file body with embedded tag pairs for expected fold regions
+     * @param mockFoldingRangesJson          the mock <code>textDocument/foldingRange</code> response
+     * @param collapsedByDefaultRangeNumbers the optional 1-based numbers of the expected ranges which should be
+     *                                       collapsed initially
+     */
     protected void assertFoldingRanges(@NotNull String fileName,
                                        @NotNull String fileBody,
-                                       @NotNull String mockFoldingRangesJson) {
+                                       @NotNull String mockFoldingRangesJson,
+                                       @NotNull Integer... collapsedByDefaultRangeNumbers) {
         MockLanguageServer.INSTANCE.setTimeToProceedQueries(100);
         List<FoldingRange> mockFoldingRanges = JSONUtils.getLsp4jGson().fromJson(mockFoldingRangesJson, new TypeToken<List<FoldingRange>>() {
         }.getType());
+        // Sort for stability
+        mockFoldingRanges.sort((foldingRange1, foldingRange2) -> {
```

Comment:
The folding ranges weren't always returned in a consistent or stable order, so I order them explicitly here so that folding ranges from LSP and fold regions from the IDE are in the same order. You'll see another sort below for the latter.